QuickFIXn/Transport/StreamFactory.cs

```diff
@@ -79,7 +81,7 @@ public static Stream CreateClientStream(IPEndPoint endpoint, SocketSettings sett
             if (!settings.SocketIgnoreProxy)
             {
                 // If system has configured a proxy for this config, use it.
-                socket = CreateTunnelThruProxy(endpoint.Address.ToString(), endpoint.Port);
+                socket = CreateTunnelThruProxy(endpoint.Address.ToString(), endpoint.Port, settings.SocketConnectHost);
```

Comment:
@IanLeeClaxton Why can't we use settings.SocketCommonName here?  Why did you need to create a new var SocketConnectHost?